tidy3d/plugins/autograd/invdes/parametrizations.py

```diff
@@ -1,46 +1,124 @@
-from typing import Tuple
+from __future__ import annotations
 
-import autograd.numpy as np
+from typing import Tuple, Union
+
+import pydantic.v1 as pd
+from numpy.typing import NDArray
+
+from tidy3d.components.base import Tidy3dBaseModel
 
 from ..types import KernelType, PaddingType
-from .filters import make_filter
+from .filters import AbstractFilter, CircularFilter, ConicFilter
 from .projections import tanh_projection
 
 
+class FilterAndProject(Tidy3dBaseModel):
+    """A class that combines filtering and projection operations.
+
+    Parameters
+    ----------
+    filter : AbstractFilter
+        The filter to apply.
+    beta : float = 1.0
+        The beta parameter for the tanh projection.
+    eta : float = 0.5
+        The eta parameter for the tanh projection.
+    """
+
+    filter: AbstractFilter
+    beta: pd.NonNegativeFloat = 1.0
```

Comment:
sort of confused why the filter is passed here but not the projector? (do we have a projector?) also would be good to include the full `pd.Field()` with the descriptions and titles there, instead of in docstring.

Reply:
Because we don't have a projector class 😆
I can make one, but right now I only made classes for stateful objects, i.e. the ones that had `make_`-functions associated with them. Since the projections are pure functions, I didn't do that for those. Do we want projector classes too?

Reply:
no it's fine, just wondering.